src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/xencoder/HaivisionXEncoderCommunicator.java

```diff
@@ -0,0 +1,813 @@
+		for (StreamStatistics streamStatistics : streamStatisticsList) {
+			String streamName = streamStatistics.getName();
+			if (EncoderConstant.NONE_STREAM_NAME.equals(streamName)) {
+				streamName = handleStreamNameIsEmpty(streamStatistics.getId());
+			}
+			String metricName = EncoderConstant.STREAM + EncoderConstant.SPACE + streamName + EncoderConstant.SPACE + EncoderConstant.STATISTICS + EncoderConstant.HASH;
+			for (StreamMonitoringMetric streamMonitoringMetric : StreamMonitoringMetric.values()) {
+				String value = getDefaultValueForNullData(streamStatistics.getValueByMetric(streamMonitoringMetric));
+				if (StreamMonitoringMetric.UPTIME.equals(streamMonitoringMetric)) {
+					stats.put(metricName + streamMonitoringMetric.getName(), formatTimeData(value));
+					continue;
+				}
+				stats.put(metricName + streamMonitoringMetric.getName(), replaceCommaByEmptyString(value));
+			}
+		}
+	}
+
+	/**
+	 * Populate audio statistics
+	 *
+	 * @param stats list statistics property
+	 */
+	private void populateAudioStatisticsData(Map<String, String> stats) {
+		for (AudioStatistics audioStatistics : audioStatisticsList) {
+			String audioName = audioStatistics.getName();
+			String metricName = audioName + EncoderConstant.SPACE + EncoderConstant.STATISTICS + EncoderConstant.HASH;
+			for (AudioMonitoringMetric audioMetric : AudioMonitoringMetric.values()) {
+				String value = getDefaultValueForNullData(audioStatistics.getValueByMetric(audioMetric));
+				if (audioMetric.equals(AudioMonitoringMetric.ENCODED_BYTES) || audioMetric.equals(AudioMonitoringMetric.ENCODED_FRAMES)) {
+					stats.put(metricName + audioMetric.getName(), replaceCommaByEmptyString(value));
+				} else {
+					stats.put(metricName + audioMetric.getName(), value);
+				}
+			}
+		}
+	}
+
+	/**
+	 * Populate video statistics
+	 *
+	 * @param stats list statistics property
+	 */
+	private void populateVideoStatisticsData(Map<String, String> stats) {
+		for (VideoStatistics videoStatistics : videoStatisticsList) {
+			String videoName = videoStatistics.getName();
+			String metricName = videoName + EncoderConstant.SPACE + EncoderConstant.STATISTICS + EncoderConstant.HASH;
+			for (VideoMonitoringMetric videoMetric : VideoMonitoringMetric.values()) {
+				String value = getDefaultValueForNullData(videoStatistics.getValueByMetric(videoMetric));
+				if (VideoMonitoringMetric.UPTIME.equals(videoMetric)) {
+					stats.put(metricName + videoMetric.getName(), formatTimeData(value));
+				} else {
+					stats.put(metricName + videoMetric.getName(), value);
+				}
+			}
+		}
+	}
+
+	/**
+	 * Replace comma by empty string
+	 *
+	 * @param value the value is string
+	 * @return value is replace the comma
+	 */
+	private String replaceCommaByEmptyString(String value) {
+		if (StringUtils.isNullOrEmpty(value)) {
+			return EncoderConstant.NONE;
+		}
+		return value.replace(EncoderConstant.COMMA, EncoderConstant.EMPTY_STRING);
+	}
+
+	/**
+	 * Format time data
```

Comment:
can you please add some information here about what the end date format is going to be?